### ISSUE TITLE
Accessibility section improvements

### ISSUE THREAD

**Issue by krisztin** [thedaviddias/Front-End-Checklist#683]
Hi there 👋️, a11y specialised dev here. The Accessibility section has a couple of issues I've collated below in a somewhat haphazard manner for now. I'd be happy to edit the readme checklist as well as the website but first wanted to open up some dialogue.

## Best practices
**Colour contrast**: Just a preference, but I'd stick to [Webaim's contrast checker](https://webaim.org/resources/contrastchecker/) as it gives more and better information

I'd add **text descriptions**: every non-decorative image should have an alt attribute describing the image itself
Good resource: [GOV.UK Accessibility Blog - Alternative text in action](https://accessibility.blog.gov.uk/2024/07/17/alternative-text-in-action/)

## Headings
**h1** - Not all pages need to have an H1, it's good practice but not a requirement by WCAG or usability. You can also use multiple h1s within a page for instance if you use each to be a header text for an individual section or article
headings - "properly and right order" is a bit vague.

For WCAG compliance and for good practice the following are true for headings:

- [ ] If text is styled to look like a heading it should be coded as such (h1-h6)
- [ ] Headings should follow a logical order e.g. `h2` news followed by `h3` (or lower) article title
- [ ] You can skip heading levels if necessary but should be avoided if possible
- [ ] They should accurately describe the section of content they head up

## Semantics

I would suggest using HTML5 tags to be used not just for input types. Use it for everything from `<nav>` to `<li>`sts, `<footer>`s and `<img>`s. Basically, `<div>`s should be a last resort.

## Forms

Here, again, whilst labelling is important, there're others:

- [ ] Radios and checkboxes - often see these mucked up. These need to be wrapped in a `<fieldset>` with a nested `<legend>` as the label
- [ ] Required fields should be marked and they should be marked visibly not just by using colour but, for instance, by adding *
- [ ] Autocomplete - for WCAG compliance you should make sure to add the appropriate autocomplete values to input fields e.g. email input should have an `autocomplete="email"` attribute
- [ ] Error handling - this is a big one from meaningful and helpful error messages to making sure to communicate errors to screen readers, there's a lot to do [Foundations: form validation and error messages](https://tetralogical.com/blog/2024/10/21/foundations-form-validation-and-error-messages/)

**Comment by Adeelmalik615343** [comment]
I can handle this

**Comment by krisztin** [comment]
@thedaviddias still open to lend a hand 😉
